### DISCUSSION TITLE
libredwg-0.12.4.4364-win64 Failed to parse AC1018 DWG drawing with error code 0xd41

### DISCUSSION THREAD

**Discussion by lym-ID** [LibreDWG/libredwg#793] [Q&A]
The error message is as follows:
Invalid opcode 0x0 in input stream at pos 294
Failed to read compressed AcDbObjects section
Failed to find section info[12] with type 2
Failed to read uncompressed AuxHeader section
Inyalid section AcDb:AppInfo count or max decompression size. Sections: 1, lax size: 0
Failed to read unconpressed AppInfo section
Some section size or address out of bounds
Failed to decode file: C:iUsers SI-PCADesktopicity bug3.dwg 0xd41

READ ERROR [Oxd41]

**Comment by rurban** [comment]
Without the dwg attached we cannot help

**Reply by lym-ID** [reply]
[libreDWG-bug-file.zip](https://github.com/LibreDWG/libredwg/files/12239722/libreDWG-bug-file.zip)

**Reply by lym-ID** [reply]
There is also a small problem: when reading the dwg file as json, the format of the color value will be confused and the following situations will occur: 1. (0-255) index value, 2.rgb (hexadecimal 8-bit code), 3.rgb (hexadecimal 8-bit code but the actual index value) for example: c3000001;  I hope the author can do a unified treatment, suggest the second.
